electrumsv/wallet_database/functions.py

```diff
@@ -1325,6 +1327,41 @@ def read_transaction_outputs(db: sqlite3.Connection, output_ids: list[Outpoint])
     return read_rows_by_ids(TransactionOutputAddRow, db, sql, sql_condition, [], output_ids)
 
 
+@replace_db_context_with_connection
+def read_transaction_fee(db: sqlite3.Connection, tx_hash: bytes) -> float | None:
```

Comment:
Please follow the instructions. The algorithm to do this is detailed in part 5. Understand it and do it. If you do not clearly understand why it does what it does or what it means to do it only for ones with sends, then ask me. We do not parse transactions in the data layer, that's just cramming workarounds for stuff we do not understand down along with database calls.